orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/DependentPipelineStarter.groovy

```diff
@@ -101,28 +102,32 @@ class DependentPipelineStarter implements ApplicationContextAware {
     //keep the trigger as the preprocessor removes it.
     def expectedArtifacts = pipelineConfig.expectedArtifacts
 
-    for (ExecutionPreprocessor preprocessor : executionPreprocessors.findAll {
-      it.supports(pipelineConfig, ExecutionPreprocessor.Type.PIPELINE)
-    }) {
-      pipelineConfig = preprocessor.process(pipelineConfig)
-    }
-
     if (parentPipelineStageId != null) {
       pipelineConfig.receivedArtifacts = artifactResolver?.getArtifacts(parentPipeline.stageById(parentPipelineStageId))
     } else {
       pipelineConfig.receivedArtifacts = artifactResolver?.getAllArtifacts(parentPipeline)
     }
 
-    pipelineConfig.trigger = trigger
-    pipelineConfig.expectedArtifacts = expectedArtifacts
-
     def artifactError = null
     try {
       artifactResolver?.resolveArtifacts(pipelineConfig)
```

Comment:
will this resolve correctly before the pipeline template is expanded?

Reply:
The artifacts should be found in the trigger, which is not a part of the pipeline template, so yes, I believe so.

Reply:
Sorry about closing and opening and resolving and unresolving. I’m on mobile and apparently pressing all the buttons 🙄

In our use case the pipeline definition _is_ the artifact, so we _have to_ resolve the artifacts before we can render the pipeline.